packages/web-integration/src/bridge-mode/agent-cli-side.ts

```diff
@@ -0,0 +1,116 @@
+import assert from 'node:assert';
+import { PageAgent } from '@/common/agent';
+import { paramStr, typeStr } from '@/common/ui-utils';
+import type { KeyboardAction, MouseAction } from '@/page';
+import {
+  BridgeUpdateAgentStatusEvent,
+  DefaultBridgeServerPort,
+} from './common';
+import { BridgeServer } from './io-server';
+import type { ChromeExtensionPageBrowserSide } from './page-browser-side';
+
+interface ChromeExtensionPageCliSide extends ChromeExtensionPageBrowserSide {
+  showStatusMessage: (message: string) => Promise<void>;
+}
+
+// actually, this is a proxy to the page in browser side
+export const getBridgePageInCliSide = (): ChromeExtensionPageCliSide => {
+  const server = new BridgeServer(DefaultBridgeServerPort);
+  server.listen();
+  const bridgeCaller = (method: string) => {
+    return async (...args: any[]) => {
+      const response = await server.call(method, args);
+      return response;
+    };
+  };
+  const page = {
+    showStatusMessage: async (message: string) => {
+      await server.call(BridgeUpdateAgentStatusEvent, [message]);
+    },
+  };
+
+  return new Proxy(page, {
+    get(target, prop, receiver) {
+      assert(typeof prop === 'string', 'prop must be a string');
+
+      if (prop === 'toJSON') {
+        return () => {
+          return {
+            pageType: 'page-over-chrome-extension-bridge',
+          };
+        };
+      }
+
+      if (prop === '_forceUsePageContext') {
+        return undefined;
+      }
+
+      if (Object.keys(page).includes(prop)) {
+        return page[prop as keyof typeof page];
+      }
+
+      if (prop === 'pageType') {
+        return 'page-over-chrome-extension-bridge';
+      }
+
+      if (prop === 'mouse') {
+        const mouse: MouseAction = {
+          click: bridgeCaller('mouse.click'),
+          wheel: bridgeCaller('mouse.wheel'),
+          move: bridgeCaller('mouse.move'),
+        };
+        return mouse;
+      }
+
+      if (prop === 'keyboard') {
+        const keyboard: KeyboardAction = {
+          type: bridgeCaller('keyboard.type'),
```

Comment:
Could these events be unified into an enum? Otherwise, it's easy to overlook new events in the future.